dlinfer/ops/llm.py

```diff
@@ -792,3 +795,34 @@ def add_rms_norm_w8a8(
     return vendor_ops_registry["add_rms_norm_w8a8"](
         hidden_states, residual, weight, epsilon, quant_dtype
     )
+
+
+def transdata_abstract_func(x: Tensor, transdata_type: int):
+    assert x.dim() in [2, 3], "x must be 2D or 3D tensor"
+    assert transdata_type == 2, "currently transdata_type must be 2"
+    assert x.dtype in [
+        torch.float16,
+        torch.int8,
+    ], "x must be float16, int8 tensor"
+    bsz = 1 if x.dim() == 2 else x.shape[0]
+    m = 16
+    n = 16 if x.dtype == torch.float16 else 32
+    res = torch.empty(
+        size=(bsz, (x.shape[-1] + n - 1) // n, (x.shape[-2] + m - 1) // m * m, n),
+        dtype=x.dtype,
+        device=x.device,
+    )
+    return res
+
+
+@register_custom_op(
+    "dlinfer::transdata",
+    ["hidden_states"],
+    impl_abstract_func=transdata_abstract_func,
+    default_value={"transdata_type": 0},
+)
+def transdata(
+    hidden_states: Tensor,
+    transdata_type: int,
+) -> Tensor:
+    return vendor_ops_registry["transdata"](hidden_states, transdata_type)
```

Comment:
increase the implementation in vendor_ops_registry["transdata"](hidden_states, transdata_type) or directly report an error